server/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -1587,8 +1590,8 @@ public long version() {
         }
 
         /**
-         * Returns {@code true} iff the get was performed from a translog operation. Notes that this returns {@code false}
-         * if the get was performed on an in-memory Lucene segment created from the corresponding translog operation.
+         * Returns {@code true} iff the get was performed from a translog operation. This possibly means that an in-memory Lucene segment
+         * needed to be created from the corresponding translog operation.
          */
         public boolean isFromTranslog() {
```

Comment:
I was still useful to keep this so that we can assert in tests that a read was actually served from the translog, but not led to the creation of an in-memory Lucene index. I've changed this by introducing another counter in InternalEngine instead, similar to how we track the number of times in-memory index needed to be created.